vendor/fuser/src/session.rs

```diff
@@ -116,6 +116,18 @@ impl<FS: Filesystem> Session<FS> {
     /// Run the session loop that receives kernel requests and dispatches them to method
     /// calls into the filesystem.
     pub fn run(&self) -> io::Result<()> {
+        self.run_and_notify(|_| {}, |_| {})
+    }
+
+    /// Run the session loop that receives kernel requests and dispatches them to method
+    /// calls into the filesystem.
+    /// This version also notifies callers of kernel requests before and after they
+    /// are dispatched to the filesystem.
+    pub fn run_and_notify<FA, FB>(&self, mut before_dispatch: FB, mut after_dispatch: FA) -> io::Result<()> 
```

Comment:
naming nit: I'd probably call it `run_with_callbacks` or something.

Reply:
reverted!